### PR TITLE
Process posting with different ASYNC_READ/BATCH_READ macro definition

### PR DESCRIPTION
**Pull request by suiguoxin**

### CODE REVIEW

File: AnnService/inc/Core/SPANN/ExtraFullGraphSearcher.h [hunk 1]
```diff
@@ -272,7 +273,37 @@ namespace SPTAG
                         LOG(Helper::LogLevel::LL_Error, "File %s read bytes, expected: %zu, acutal: %llu.\n", m_extraFullGraphFile.c_str(), totalBytes, numRead);
                         exit(-1);
                     }
-                    ProcessPosting(m_vectorInfoSize)
+                    // decompress posting list
+                    char* p_postingListFullData = buffer + listInfo->pageOffset;
+                    if (m_enableDataCompression)
```
Comment (MaggieQi) [inline AnnService/inc/Core/SPANN/ExtraFullGraphSearcher.h:278]: too many repeated code. Could we move the decompression to marco or use a function to reduce the repeated code?

**Review by MaggieQi — CHANGES_REQUESTED** [review]

File: AnnService/inc/Core/SPANN/ExtraFullGraphSearcher.h (hunk 1 above)
Comment (suiguoxin) [reply AnnService/inc/Core/SPANN/ExtraFullGraphSearcher.h:272]: > too many repeated code. Could we move the decompression to marco or use a function to reduce the repeated code?

refactored

File: AnnService/inc/Core/SPANN/ExtraFullGraphSearcher.h [hunk 2]
```diff
@@ -192,6 +211,7 @@ namespace SPTAG
                         fileid = curPostingID / m_listPerFile;
                         listInfo = &(m_listInfos[fileid][curPostingID % m_listPerFile]);
                     }
+                    listInfo->postingID = curPostingID;
```
Comment (MaggieQi) [inline AnnService/inc/Core/SPANN/ExtraFullGraphSearcher.h:214]: this information can be assigned in the LoadHeadInfo

Comment (suiguoxin) [reply AnnService/inc/Core/SPANN/ExtraFullGraphSearcher.h]: Done

File: AnnService/inc/Core/SPANN/ExtraFullGraphSearcher.h [hunk 3]
```diff
@@ -656,6 +662,8 @@ namespace SPTAG
                         postingListBytes[i] = postingListSize[i] * vectorInfoSize;
                     }
                 }
+
+                if (p_opt.m_ssdIndexFileNum > 1) selections.SaveBatch();
```
Comment (MaggieQi) [inline AnnService/inc/Core/SPANN/ExtraFullGraphSearcher.h:666]: why move this line after fullVectors read? This causes the memory overflow issue in the batch build.

Comment (suiguoxin) [reply AnnService/inc/Core/SPANN/ExtraFullGraphSearcher.h]: Fixed

**Review by MaggieQi — APPROVED** [review]

**Pull request merged**